extensions/azure/cosmos/contract-negotiation-store-cosmos/src/test/java/org/eclipse/dataspaceconnector/contract/negotiation/store/CosmosContractNegotiationStoreIntegrationTest.java

```diff
@@ -420,7 +420,7 @@ void findAll_verifyFiltering_invalidFilterExpression() {
 
         var query = QuerySpec.Builder.newInstance().filter("something contains other").build();
 
-        assertThatThrownBy(() -> store.queryNegotiations(query)).isInstanceOfAny(IllegalArgumentException.class).hasMessage("Cannot build SqlParameter for operator: contains");
+        assertThatThrownBy(() -> store.queryNegotiations(query)).isInstanceOfAny(IllegalArgumentException.class).hasMessage("Cannot build WHERE clause, reason: unsupported operator contains");
```

Comment:
In general I would avoid testing exception messages, especially in integration tests

Reply:
Well to me it seems a good idea, so we ensure the exception we are testing for is not masked by an unrelated one

Reply:
I tend to agree with @algattik, and asserting exception message has already surfaced problems for me on several occasions. I like tests to be as rigid as possible - you can never be too certain :)